### PR TITLE
TypeScript Support

### PR DESCRIPTION
**Pull request by evanshortiss**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case
TypeScript users will be able to use this and benefit from compile time checks. They can either import the module:

```
import NoSleep from 'nosleep.js'

const slp = new NoSleep()
```

or just the typings if using a CDN copy:

```
import 'nosleep.js`

const slp = new window.NoSleep()
```

### Breaking Changes
N/A

### Additional Info
This can also be hosted on DefinitelyTyped and installed via `npm install @types/nosleep.js` if you don't want to merge here, it just helps to include types in the core module.

### CODE REVIEW

**Comment by evanshortiss** [comment]
@fgerschau thanks for approving. Any plan to merge these?

**Comment by fgerschau** [comment]
@evanshortiss Unfortunately I don't have the rights to do so but I'm interested in having your changes merged as well, that's why I reviewed them 😄 
@richtr Could you please have a look?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nosleep.js",
-  "version": "0.9.0",
+  "version": "0.10.0",
```
Comment (richtr) [inline package.json:3]: Would you be able to remove this version bump from the PR? 

I plan to then merge this PR with a version bump happening in a separate (collective) way.

**Comment by richtr** [comment]
Since this is an old PR and the original developer may not be available I have gone ahead and duplicated this PR, removing the version bump. 

Closing this PR in favor of: https://github.com/richtr/NoSleep.js/pull/93.

**Comment by evanshortiss** [comment]
@richtr thanks!

**Pull request closed without merging**